diff --git a/lib/app/pages/subscription/subscription_view.dart b/lib/app/pages/subscription/subscription_view.dart
index 0e3535e..10a701b 100644
--- a/lib/app/pages/subscription/subscription_view.dart
+++ b/lib/app/pages/subscription/subscription_view.dart
@@ -19,6 +19,7 @@
   @override
   Widget get view {
     return Scaffold(
+      key: globalKey,
       appBar: PreferredSize(
         preferredSize: Size.fromHeight(
           Theme.of(context).appBarTheme.toolbarHeight ?? 74.0,