code
Diff
download
content_copy
expand_less

diff --git a/lib/app/theme/dynamic_theme.dart b/lib/app/theme/dynamic_theme.dart
index 1014b77..091515c 100644
--- a/lib/app/theme/dynamic_theme.dart
+++ b/lib/app/theme/dynamic_theme.dart
@@ -19,6 +19,9 @@
           color: Color(0xFFe98c14),
         ),
       ),
+      progressIndicatorTheme: const ProgressIndicatorThemeData(
+        color: Color(0xFFe98c14),
+      ),
       focusColor: Colors.blueGrey[900],
       appBarTheme: AppBarTheme(
         elevation: 0,
@@ -73,6 +76,9 @@
 
   static ThemeData getDark(Settings settings) {
     return ThemeData(
+      progressIndicatorTheme: const ProgressIndicatorThemeData(
+        color: Color(0xFFf49315),
+      ),
       dividerColor: Colors.transparent,
       focusColor: Colors.blueGrey[900],
       shadowColor: const Color(0xFF353535),