diff --git a/app/pages/notes/notes_controller.dart b/app/pages/notes/notes_controller.dart
index 9464019..9180708 100644
--- a/app/pages/notes/notes_controller.dart
+++ b/app/pages/notes/notes_controller.dart
@@ -161,22 +161,22 @@
                 child: pw.Column(
                   crossAxisAlignment: pw.CrossAxisAlignment.start,
                   children: [
+                    pw.Text(
+                      TextUtils.parseHtmlString(note.content),
+                      style: pw.TextStyle(font: ttfFont, fontSize: 14),
+                    ),
+                    pw.SizedBox(height: 10),
+                    pw.Divider(color: PdfColors.grey400, height: 15),
                     pw.Text(
                       'Документ: ${note.regulationTitle}',
                       style: pw.TextStyle(
                           font: ttfFont,
                           fontSize: 10,
                           fontWeight: pw.FontWeight.bold),
                     ),
                     pw.Text(
                       'Глава: ${note.chapterName}',
                       style: pw.TextStyle(
                           font: ttfFont,
                           fontSize: 10,
                           fontWeight: pw.FontWeight.bold),
                     ),
-                    pw.Divider(color: PdfColors.grey400, height: 15),
-                    pw.Text(
-                      TextUtils.parseHtmlString(note.content),
-                      style: pw.TextStyle(font: ttfFont, fontSize: 14),
-                    ),
-                    pw.SizedBox(height: 10),
                   ],
                 ),
               );
diff --git a/app/pages/notes/notes_view.dart b/app/pages/notes/notes_view.dart
index 004113e..2d2744c 100644
--- a/app/pages/notes/notes_view.dart
+++ b/app/pages/notes/notes_view.dart
@@ -87,18 +87,28 @@
             itemBuilder: (BuildContext context) => <PopupMenuEntry<String>>[
               if (controller.hasNotes)
                 PopupMenuItem<String>(
                   value: 'export',
                   child: ListTile(
-                    leading: const Icon(Icons.picture_as_pdf_outlined),
-                    title: const Text('Экспорт в PDF'),
+                    leading: Icon(Icons.picture_as_pdf_outlined,
+                        color: Theme.of(context).textTheme.bodyLarge?.color),
+                    title: Text('Экспорт в PDF',
+                        style: Theme.of(context).textTheme.bodyLarge),
                     contentPadding: EdgeInsets.zero,
                   ),
                 ),
               PopupMenuItem<String>(
                 value: 'sort',
                 child: ListTile(
-                  leading: const Icon(Icons.sort),
-                  title: const Text('Сортировать'),
+                  leading: Icon(Icons.sort,
+                      color: Theme.of(context).textTheme.bodyLarge?.color),
+                  title: Text('Сортировать',
+                      style: Theme.of(context).textTheme.bodyLarge),
                   contentPadding: EdgeInsets.zero,
                 ),
               ),