diff --git a/lib/app/pages/exam/exam_view.dart b/lib/app/pages/exam/exam_view.dart
index 018131e..e00495f 100644
--- a/lib/app/pages/exam/exam_view.dart
+++ b/lib/app/pages/exam/exam_view.dart
@@ -147,23 +147,33 @@
   }
 
   Widget _buildAnswerTile(ExamController controller, String answer, ExamQuestion question) {
-    bool isSelected = controller.selectedAnswer == answer;
-    bool isCorrect = controller.isCorrect(answer, question);
+    final bool isSelected = controller.selectedAnswer == answer;
+    final bool isThisAnswerCorrect = controller.isCorrect(answer, question);
+    final bool wasUserSelectionCorrect =
+        controller.isCorrect(controller.selectedAnswer ?? '', question);
+
     Color? tileColor;
+    Icon? trailingIcon;
 
     if (controller.isAnswered) {
-      if (isCorrect) {
-        tileColor = Colors.green.withOpacity(0.2);
-      } else if (isSelected && !isCorrect) {
-        tileColor = Theme.of(context).colorScheme.error.withOpacity(0.2);
+      if (isSelected) {
+        if (isThisAnswerCorrect) {
+          // User picked this correct answer
+          tileColor = Colors.green.withOpacity(0.2);
+          trailingIcon = const Icon(Icons.check_circle, color: Colors.green);
+        } else {
+          // User picked this wrong answer
+          tileColor = Theme.of(context).colorScheme.error.withOpacity(0.2);
+          trailingIcon =
+              Icon(Icons.cancel, color: Theme.of(context).colorScheme.error);
+        }
+      } else if (isThisAnswerCorrect && !wasUserSelectionCorrect) {
+        // This is a correct answer, and the user picked a wrong one. Show the correct answer.
+        tileColor = Colors.green.withOpacity(0.2);
+        trailingIcon =
+            const Icon(Icons.check_circle_outline, color: Colors.green);
       }
     }
 
     return Card(
       elevation: 0,
       color: tileColor ?? Theme.of(context).scaffoldBackgroundColor,
       margin: EdgeInsets.zero,
       shape: Border(
         bottom: BorderSide(
           width: 1.0,
           color: Theme.of(context).shadowColor,
         ),
       ),
       child: ListTile(
         title: Text(answer),
         onTap: () => controller.selectAnswer(answer),
-        trailing: controller.isAnswered
-            ? isCorrect
-                ? const Icon(Icons.check_circle, color: Colors.green)
-                : (isSelected
-                    ? Icon(Icons.cancel, color: Theme.of(context).colorScheme.error)
-                    : null)
-            : null,
+        trailing: trailingIcon,
       ),
     );
   }