diff --git a/lib/app/app.dart b/lib/app/app.dart
index 9142823..012465c 100644
--- a/lib/app/app.dart
+++ b/lib/app/app.dart
@@ -30,6 +30,7 @@
         settingsRepository: settingsRepository,
         ttsRepository: ttsRepository,
         notesRepository: notesRepository,
+        subscriptionRepository: subscriptionRepository,
       ),
       debugShowCheckedModeBanner: false,
     );
diff --git a/lib/app/pages/main/main_presenter.dart b/lib/app/pages/main/main_presenter.dart
index 285433a..0884813 100644
--- a/lib/app/pages/main/main_presenter.dart
+++ b/lib/app/pages/main/main_presenter.dart
@@ -1,5 +1,6 @@
 import 'package:flutter_clean_architecture/flutter_clean_architecture.dart';
 import '../../../domain/repositories/regulation_repository.dart';
+import '../../../domain/repositories/subscription_repository.dart';
 import '../../../domain/repositories/settings_repository.dart';
 import '../../../domain/repositories/tts_repository.dart';
 import '../../../domain/repositories/notes_repository.dart';
@@ -18,6 +19,7 @@
     required SettingsRepository settingsRepository,
     required TTSRepository ttsRepository,
     required NotesRepository notesRepository,
+    required SubscriptionRepository subscriptionRepository,
   }) : _regulationRepository = regulationRepository;
   // _settingsRepository = settingsRepository,
   // _ttsRepository = ttsRepository,
diff --git a/lib/app/pages/notes/notes_view.dart b/lib/app/pages/notes/notes_view.dart
index a94164a..5695024 100644
--- a/lib/app/pages/notes/notes_view.dart
+++ b/lib/app/pages/notes/notes_view.dart
@@ -24,6 +24,7 @@
     return ControlledWidgetBuilder<NotesController>(
       builder: (context, controller) {
         return Scaffold(
+          key: globalKey,
           appBar: PreferredSize(
             preferredSize: Size.fromHeight(
               Theme.of(context).appBarTheme.toolbarHeight ?? 74.0,