diff --git a/lib/app/pages/exam/exam_controller.dart b/lib/app/pages/exam/exam_controller.dart
index 813134e..838234e 100644
--- a/lib/app/pages/exam/exam_controller.dart
+++ b/lib/app/pages/exam/exam_controller.dart
@@ -62,12 +62,7 @@
   final DataRegulationRepository _dataRepository = DataRegulationRepository();
   ExamController(
     this.regulationId,
     this.isSubscribed,
-  ) : _presenter = ExamPresenter(CloudExamRepository()) {
-    if (!isSubscribed) {
-      _isTrainingMode = true;
-    }
+  ) : _presenter = ExamPresenter(CloudExamRepository()) {
     initListeners();
     _loadSettingsAndGetQuestions();
   }
diff --git a/lib/app/pages/exam/exam_view.dart b/lib/app/pages/exam/exam_view.dart
index 673693e..1754406 100644
--- a/lib/app/pages/exam/exam_view.dart
+++ b/lib/app/pages/exam/exam_view.dart
@@ -6,7 +6,7 @@
 
 class ExamArguments {
   final int regulationId;
-  final bool isSubscribed;
-  ExamArguments({required this.regulationId, this.isSubscribed = true});
+  final bool isSubscribed;
+  ExamArguments({required this.regulationId, this.isSubscribed = true});
 }
 
 class ExamView extends View {
@@ -325,12 +325,32 @@
               mainAxisAlignment: MainAxisAlignment.center,
               crossAxisAlignment: CrossAxisAlignment.stretch,
               children: [
-                Text('Доступен только для подписчиков',
-                    textAlign: TextAlign.center,
-                    style: Theme.of(context).textTheme.headlineSmall),
+                Text(
+                  'Доступен только для подписчиков',
+                  textAlign: TextAlign.center,
+                  style: Theme.of(context).textTheme.headlineSmall?.copyWith(
+                        color: Theme.of(context).textTheme.bodyLarge?.color,
+                      ),
+                ),
                 const SizedBox(height: 24),
                 ElevatedButton(
-                    onPressed: () =>
-                        Navigator.of(context).pushNamed('/subscription'),
-                    child: const Text('Оформить подписку')),
+                  style: ElevatedButton.styleFrom(
+                    backgroundColor: Theme.of(context)
+                        .navigationRailTheme
+                        .selectedIconTheme
+                        ?.color,
+                    foregroundColor: Colors.white,
+                    padding: const EdgeInsets.symmetric(vertical: 16),
+                    shape: RoundedRectangleBorder(
+                      borderRadius: BorderRadius.circular(12),
+                    ),
+                    textStyle: Theme.of(context).textTheme.bodyLarge?.copyWith(
+                          color: Colors.white,
+                        ),
+                  ),
+                  onPressed: () =>
+                      Navigator.of(context).pushNamed('/subscription'),
+                  child: const Text('Оформить подписку'),
+                ),
               ],
             ),
           ),